frida-gum/src/stalker/transformer.rs

```diff
@@ -1,6 +1,47 @@
 use crate::{CpuContext, Gum};
 use std::marker::PhantomData;
-use std::os::raw::c_void;
+use std::collections::HashMap;
+use std::ffi::c_void;
+
+//use std::rc::Rc;
+use std::cell::RefCell;
+
+//use lazy_static::lazy_static;
+
+struct ClosureMap {
+    cache: HashMap<*const c_void, *mut c_void>,
+}
+
+impl ClosureMap {
+    pub fn new() -> Self {
+        Self {
+            cache: HashMap::new(),
+        }
+    }
+    pub fn contains_key(&self, key: *const c_void) -> bool {
+        //println!("contains_key: {:?}", key);
+        self.cache.contains_key(&key)
+    }
+
+    pub fn insert(&mut self, key: *const c_void, value: *mut c_void) -> Option<*mut c_void> {
+        self.cache.insert(key, value)
+    }
+
+    pub fn get(&self, key: *const c_void) -> *mut c_void {
```

Comment:
This should be an `Option<*mut c_void>` so that there is no `unwrap`. Then we don't need the contains_key check beforehand. This is more Rust-like.